Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -224,6 +224,34 @@ final class ConnectionPool {
         fileprivate var closePromise: EventLoopPromise<Void>
 
         var closeFuture: EventLoopFuture<Void>
+
+        func removeIdleConnectionHandlersForLease() -> EventLoopFuture<Connection> {
+            return self.channel.eventLoop.flatSubmit {
+                self.removeHandler(IdleStateHandler.self).flatMap { () -> EventLoopFuture<Bool> in
+                    self.channel.pipeline.handler(type: IdlePoolConnectionHandler.self).flatMap { idleHandler in
+                        self.channel.pipeline.removeHandler(idleHandler).flatMapError { _ in
+                            self.channel.eventLoop.makeSucceededFuture(())
+                        }.map {
+                            idleHandler.timeoutClosed.load()
```

Comment:
you need to _also_ check `self.channel.isActive` here. The close could've come from either us (checked with `idleHandler.timeoutClosed`) or the other side (to be checked with `self.channel.isActive`)

Reply:
Thanks, fixed it!